torch/csrc/jit/serialization/storage_context.h

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include <ATen/core/ivalue.h>
+
+namespace torch {
+namespace jit {
+
+// Used in torch.package and TorchScript serialization to coordinate
+// sharing of storages between models. Also used to create deterministic
+// naming for storages.
+class TORCH_API SerializationStorageContext {
+ public:
+  explicit SerializationStorageContext() : unique_id(0) {}
+
+  uint64_t addStorage(c10::Storage storage) {
```

Comment:
Seems like the API would be cleaner with just `getOrAddStorage`, which either retrieves an id from the table or creates a new one. Some benefits:
1. It is no longer possible to use the API incorrectly (e.g. those `TORCH_INTERNAL_ASSERT`s you have now will no longer need to exist). It's nice to make APIs that are impossible to use incorrectly.
2. It simplifies user code (you can get rid of the if-else branch in export_module.cpp:477)